messaging/src/main/java/org/axonframework/eventhandling/tokenstore/jpa/JpaTokenStore.java

```diff
@@ -118,8 +118,31 @@ public void initializeTokenSegments(String processorName, int segmentCount, Trac
     @Override
     public void storeToken(TrackingToken token, String processorName, int segment) {
         EntityManager entityManager = entityManagerProvider.getEntityManager();
-        TokenEntry tokenEntry = loadToken(processorName, segment, entityManager);
-        tokenEntry.updateToken(token, serializer);
+        TokenEntry tokenToStore = new TokenEntry(processorName, segment, token, serializer);
+        byte[] tokenDataToStore =
+                getOrDefault(tokenToStore.getSerializedToken(), SerializedObject::getData, new byte[0]);
+        String tokenTypeToStore = getOrDefault(tokenToStore.getTokenType(), SerializedType::getName, null);
+
+        int updatedTokens = entityManager.createQuery("UPDATE TokenEntry te SET "
+                                                              + "te.token = :token, "
+                                                              + "te.tokenType = :tokenType, "
+                                                              + "te.timestamp = :timestamp "
+                                                              + "WHERE te.owner = :owner "
+                                                              + "AND te.processorName = :processorName "
+                                                              + "AND te.segment = :segment")
+                                         .setParameter("token", tokenDataToStore)
+                                         .setParameter("tokenType", tokenTypeToStore)
+                                         .setParameter("timestamp", tokenToStore.timestampAsString())
+                                         .setParameter("owner", nodeId)
+                                         .setParameter("processorName", processorName)
+                                         .setParameter("segment", segment)
+                                         .executeUpdate();
+
+        if (updatedTokens == 0) {
+            // Update couldn't succeed, trying to first load the token and than update it instead.
```

Comment:
Is it useful to `log.debug` for example when it happens?

Reply:
Sure thing, added!